File: figures/src/python/build_volcano_pathway_figure.py
#!/usr/bin/env python3
"""Compatibility wrapper for the volcano/pathway figure."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys


REPO_ROOT = Path(__file__).resolve().parents[3]


def main() -> int:
    subprocess.run(
        [
            sys.executable,
            str(REPO_ROOT / "figures/src/python/run_class_renderer.py"),
            "--class",
            "volcano_pathway_compound",
            "--spec",
            "figures/specs/figure_02_volcano_pathway.yml",
        ],
        cwd=REPO_ROOT,
        check=True,
    )
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: figures/src/r/build_volcano_pathway_figure.R
#!/usr/bin/env Rscript

script_arg <- grep("^--file=", commandArgs(trailingOnly = FALSE), value = TRUE)
if (length(script_arg) == 0) {
  stop("Unable to locate the current R script path")
}
script_path <- normalizePath(sub("^--file=", "", script_arg[[1]]), winslash = "/", mustWork = TRUE)
repo_root <- normalizePath(file.path(dirname(script_path), "..", "..", ".."), winslash = "/", mustWork = TRUE)
options(manuscript.figure.repo_root = repo_root)

source(file.path(repo_root, "figures", "src", "r", "classes", "volcano_pathway_compound.R"))
build_figure(normalizePath(file.path(repo_root, "figures/specs/figure_02_volcano_pathway.yml"), winslash = "/", mustWork = TRUE))
